### PR TITLE
Generic api errors, delete staging files on job delete

### LINKED ISSUE #174: address open concerns in #146 (output files PR)
See discussion in https://github.com/jupyter-server/jupyter-scheduler/pull/146

### LINKED ISSUE #216: Show more specific job creation errors in UI
### Problem

All job creation errors are returned from the API, and displayed in the UI, as `Unexpected error occurred during creation of Job`.

https://github.com/jupyter-server/jupyter-scheduler/blob/682ddbe33219cc08d997c7e9b81bc3768dac8fac/jupyter_scheduler/handlers.py#L103-L107

### Proposed Solution

Users should see more specific, meaningful error messages in the UI (as returned from the API).

### Additional context

Should we display the exact message or a series of more specific generic messages?

### PR DESCRIPTION
**Pull request by 3coins**
Fixes #174 
Fixes #216

### CODE REVIEW

File: jupyter_scheduler/exceptions.py [hunk 1]
```diff
@@ -1,3 +1,11 @@
+class SchedulerError(Exception):
```
Comment (ellisonbg) [inline jupyter_scheduler/exceptions.py:1]: Will this enable a backend to raise a `SchedulerError` with a custom message and have that sent back to the frontend to display the error message in the create form banner?

File: jupyter_scheduler/scheduler.py [hunk 2]
```diff
@@ -473,6 +474,12 @@ def delete_job(self, job_id: str):
             if Status(job_record.status) == Status.IN_PROGRESS:
                 self.stop_job(job_id)
 
+            staging_paths = self.get_staging_paths(DescribeJob.from_orm(job_record))
```
Comment (ellisonbg) [inline jupyter_scheduler/scheduler.py:477]: This looks unrelated, is this a fix for something else?

File: jupyter_scheduler/exceptions.py (hunk 1 above)
Comment (3coins) [reply jupyter_scheduler/exceptions.py:1]: Correct.

File: jupyter_scheduler/scheduler.py (hunk 2 above)
Comment (3coins) [reply jupyter_scheduler/scheduler.py:477]: This was raised by @dlq in a previous PR.
https://github.com/jupyter-server/jupyter-scheduler/pull/146#discussion_r998658523

**Review by dlqqq — APPROVED** [review]
Looks great!
